### ISSUE TITLE
PWG render has gray background

### ISSUE THREAD

**Issue by sesu-bio** [OpenPrinting/cups-filters#470]
I noticed that when printing using a driverless queue, some documents have a gray background (it should be white). This is reproducable on several printers and happens consistently with this type of document. We use PostScript files as the input for CUPS. This is one of the problematic files:
[test.zip](https://github.com/OpenPrinting/cups-filters/files/8878504/test.zip)

When I manually convert the file to PWG raster and look at the output in rasterview, I can see that the background is gray.
`cupsfilter -m image/pwg-raster ./print_link_1.ps > test.pwg`
[stdout.txt](https://github.com/OpenPrinting/cups-filters/files/8878513/stdout.txt)
[pwg.zip](https://github.com/OpenPrinting/cups-filters/files/8878521/pwg.zip)
![screenshot](https://user-images.githubusercontent.com/73291406/173058232-e5d2ff65-1c7e-45e1-b5b2-7b20b7ec37e1.PNG)

When I convert the file to PDF using cupsfilter, the background looks fine in a PDF viewer. But when I convert that pdf to pwg, it is gray again.
[test.pdf](https://github.com/OpenPrinting/cups-filters/files/8878538/test.pdf)

When I try to convert the PostScript file to Apple raster using cupsfilter, rasterview cannot open it at all. cupfilter prints an error message.
`cupsfilter -m image/pwg-raster ./print_link_1.ps > test.pwg`
`ERROR: Unsupported raster data.`
[urf.zip](https://github.com/OpenPrinting/cups-filters/files/8878548/urf.zip)
[stdout_urf.txt](https://github.com/OpenPrinting/cups-filters/files/8878573/stdout_urf.txt)

The version of cups-filters used is 1.28.10.

**Comment by sesu-bio** [comment]
I forgot to mention: The PostScript file is generated by the "Microsoft PS Class Driver" on Windows. When printed using a raw queue, the background is white as expected.

**Comment by sesu-bio** [comment]
I tried to reproduce the problem on several other systems.
- Raspberry Pi OS 11 with cups-filters 1.28.7 and ghostscript 9.53.3: **not reproducable**
- Ubuntu 20.04 with cups-filters 1.27.4 and ghostscript 9.50: **not reproducable** (although the logo and table headers are "dotted" instead of properly filled)
- EndeavourOS with cups-filters 1.28.15 ghostscript 9.56.1: **not reproducable**

It seems to be a problem specific to our Linux System which is based on BuildRoot 2022.02 and uses cups 2.4.0, cups-filters 1.28.10 and ghostscript 9.55.0. Possibly a missing dependency or a bad combination.

**Comment by sesu-bio** [comment]
When invoking GhostScript directly using the same parameters as cupsfilter, the issue can be reproduced:

```
gs -dSAFER -dNOPAUSE -dBATCH -dNOINTERPOLATE -dNOMEDIAATTRS -sOutputFile=./gs.pdf -sDEVICE=pwgwrite -dDoNumCopies -dCompatibilityLevel=1.3 -dAutoRotatePages=/None -dAutoFilterColorImages=false -dNOPLATFONTS -dColorImageFilter=/FlateEncode -dPDFSETTINGS=/default -dColorConversionStrategy=/LeaveColorUnchanged -r600x600 -dDEVICEWIDTHPOINTS=612 -dDEVICEHEIGHTPOINTS=792 -I/usr/share/cups/fonts ./test.ps

gs -dSAFER -dNOPAUSE -dBATCH -dNOINTERPOLATE -dNOMEDIAATTRS -dShowAcroForm -sOutputFile=./gs.pwg -sDEVICE=cups -sMediaClass=PwgRaster -sOutputType=Automatic -r600x600 -dDEVICEWIDTHPOINTS=612 -dDEVICEHEIGHTPOINTS=792 -dcupsBitsPerColor=8 -dcupsColorOrder=0 -dcupsColorSpace=19 -dcupsBorderlessScalingFactor=0.0000 -dcupsInteger0=1 -dcupsInteger1=1 -dcupsInteger2=1 -scupsPageSizeName=na_letter_8.5x11in -I/usr/share/cups/fonts ./gs.pdf
```

Changing the dcupsColorSpace parameter to a grayscale one (3 or 18) produces an output with white background. Most of the other ones I tested produce the gray background.

**Comment by tillkamppeter** [comment]
Which Ghostscript version is that? I do not remember that there was ever one with a "pwgwrite" output device. I had actually called it "pwgraster" by the way.

**Comment by tillkamppeter** [comment]
According to the other command line options the first Ghostscript line should probably be the one to turn your PostScript input file into PDF and so creating the input file for the second line. The correct output device for that is "pdfwrite". Please copy and paste these lines and do not write them from the top of your head, so that we can also actually reproduce your problem.

**Comment by sesu-bio** [comment]
The Ghostscript version is 9.55.0.
Sorry about the incorrect command. This was something I tried before but didnt work for obvious reasons. I edited my comment above to contain the commands I can reproduce the problem with.

**Comment by tillkamppeter** [comment]
I have now repeated the test with your corrected command lines, at least for Ghostscript 9.55.0 and I still get the correct, white background.

**Comment by tillkamppeter** [comment]
Not really, looks like a bug in a particular distribution's package. Please check on which distribution and which release of it you can reproduce this bug and report a bug to there. Please post the link to that report here.

**Comment by sesu-bio** [comment]
I found a similar issue [here](https://github.com/ImageMagick/ImageMagick/discussions/4797) but it looks like nobody found the root cause.
Is this likely an issue with Ghostscript only or does Ghostscript use something from CUPS to render the files? I am asking because some of the parameters sound very CUPS specific.

**Comment by tillkamppeter** [comment]
It is principally only Ghostscript, as you can reproduce it with only Ghostscript command lines. The CUPS-specific parameters are due to the fact that the rasterizing (second command line) is done with the CUPS output device.

**Comment by sesu-bio** [comment]
Thank you for your help. I will link the bug report.

**Comment by sesu-bio** [comment]
I figured out what the root cause is. GhostScript contains sources for a number of its dependencies in its own source code. When building GhostScript, BuildRoot deletes these sources which results in GhostScript using the libraries from the system instead. One of these libraries is [Little-CMS](https://www.littlecms.com/). The current LTS release of BuildRoot contains Little-CMS 2.13 which has a [known issue with gray backgrounds](https://github.com/mm2/Little-CMS/commit/fdbfb7694f9d7048d53674b79ddfc38068bfdaf7). It is fixed in [version 2.13.1](https://github.com/mm2/Little-CMS/releases/tag/lcms2.13.1).
I did not open an issue at BuildRoot because they already updated to the fixed version of Little-CMS. The fix has not been included in the latest LTS release 2022.02.3 but should land in the next one.
